### PR TITLE
[Telemetry] Update system metrics.

### PR DESCRIPTION
**Pull request by JoshLind**
## Motivation

This PR updates the system information sent by telemetry. Specifically, we:
1. Ensure that system information is being collected on node startup.
2. Rename the "/json_metrics" API to "/system_information". This makes more sense, e.g., `http://127.0.0.1:9101/system_information` now shows system information 😄
3. Extend the system metrics/information to include more CPU, disk and memory information.
4. Clean up and better document the code.

Note: the accuracy of the `sysinfo` crate is still to be determined as the information reported on my machine isn't 100% correct... 🤔 

Next steps are to refactor the telemetry APIs, build services with configurable push frequencies and start pushing the relevant metrics for better observability 😄 

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Yes.

## Test Plan

Manually started a fullnode and watched the `metrics` and `system_information` APIs. Example of hitting the `system_information` endpoint:

```
{"cpu":"{\"cpu_brand\":\"Apple M1 Max\",\"cpu_frequency\":\"0\",\"cpu_name\":\"0\",\"cpu_vendor_id\":\"Apple\"}","cpu_core_count":"10",
"cpu_count":"10",
"disk":"{\"disk_available_space\":\"793433276416\",\"disk_file_system\":\"[97, 112, 102, 115]\",\"disk_name\":\"\\\"VM\\\"\",\"disk_total_space\":\"994662584320\",\"disk_type\":\"SSD\"}","
disk_count":"9",
"system":"{\"system_host_name\":\"MacBook-Pro.local\",\"system_kernel_version\":\"21.2.0\",\"system_name\":\"Darwin\",\"system_os_version\":\"MacOS 12.1 \"}"}
```

## Related PRs

None.

### CODE REVIEW

File: crates/aptos-metrics/src/system_information.rs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+fn collect_git_revision(system_information: &mut HashMap<String, String>) {
+    let git_rev = get_git_rev();
+    if !git_rev.is_empty() {
+        system_information.insert(GIT_REVISION.to_string(), git_rev);
+    }
+}
+
+/// Fetches the git revision from the environment variables
+pub fn get_git_rev() -> String {
+    env!("GIT_REV").to_string()
```
Comment (rustielin) [inline crates/aptos-metrics/src/system_information.rs:46]: I was planning to use `shadow.rs` for this. See how the CLI does it https://github.com/aptos-labs/aptos-core/blob/85425247c7306c741cc73a1d8b6fe8a82060fd42/crates/aptos/src/common/utils.rs#L93

Comment (JoshLind) [reply crates/aptos-metrics/src/system_information.rs:67]: Also, I've updated the PR description with an example of fetching the system information on my mac. Mac information isn't super accurate when using the sys info crate. But, I figured an example couldn't hurt 😄

**Review by rustielin — APPROVED** [review]
Overall LGTM! Thanks for taking this on @JoshLind 

Just left a nit wrt unifying the build info we're collecting for `aptos-node` with the `aptos` CLI, but stamping for now to not block

File: crates/aptos-metrics/src/system_information.rs (hunk 1 above)
Comment (JoshLind) [reply crates/aptos-metrics/src/system_information.rs:67]: Awesome, thanks @rustielin! Yes, for sure. Once I have the schema stable, I'll write up an API document and update the dev docs!

File: crates/aptos-metrics/src/system_information.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+}
+
+// TODO(joshlind): find a way of removing this.
+/// Fetches the git revision from the environment variables
+pub fn get_git_rev() -> String {
+    env!("GIT_REV").to_string()
+}
+
+/// Collects the build info and appends it to the given map
+fn collect_build_info(system_information: &mut BTreeMap<String, String>) {
```
[the remark this answers is not in the document]
Comment (JoshLind) [reply crates/aptos-metrics/src/system_information.rs:71]: Yes, exactly. I want one function that fetches the same information and we can call it from both 😄

**Comment by JoshLind** [comment]
Thanks @rustielin!

/land

**Pull request closed without merging**